scipy/spatial/ckdtree.pyx

```diff
@@ -1076,46 +1117,173 @@ cdef public class cKDTree [object ckdtree, type ckdtree_type]:
+            the edges of the bins, and must be non-decreasing.
         p : float, optional 
-            1<=p<=infinity, default 2.0
-            Which Minkowski p-norm to use
+            1<=p<=infinity. 
+            Which Minkowski p-norm to use.
+            Default 2.0.
+        weights : tuple, array_like, or None, optional
+            If given as a tuple, weights[0] is the weights of points in self, and
+            weights[1] is the weights of points in other; either can be None to 
+            indicate the points are unweighted.
+            If given as an array_like, weights is the weights of points in self 
+            and other. If self and other are two different trees, an Error is raised.
+            If None, the pair-counting is unweighted.
+            Default: None
+        cumulative : bool, optional
+            Whether the returned counts are cumulative. When cumulative is set to ``False``
+            the algorithm is optimized to work with a large number of bins (>10) specified
+            by ``r``. When ``cumulative`` is set to True, the algorithm is optimized to work
+            with a small number of ``r``. Default: True
 
         Returns
         -------
-        result : int or 1-D array of ints
-            The number of pairs.
+        result : scalar or 1-D array
+            The number of pairs. For unweighted counts, the result is integer.
+            For weighted counts, the result is float.
+            If cumulative is False, ``result[i]`` contains the counts with
+            ``(-inf if i == 0 else r[i-1]) < R <= r[i]``
+
+        Notes
+        -----
+
+        Pair-counting is the basic operation used to calculate the two point
+        correlation functions from a data set composed of position of objects.
+
+        Two point correlation function measures the clustering of objects and
+        is widely used in cosmology to quantify the large scale structure
+        in our Universe, but it may be useful for data analysis in other fields
+        where self-similar assembly of objects also occur.
+
+        The Landy-Szalay estimator for the two point correlation function of
+        ``D`` measures the clustering signal in ``D``. [2]_
+
+        For example, given the position of two sets of objects,
+
+        - objects ``D`` (data) contains the clustering signal, and
+
+        - objects ``R`` (random) that contains no signal,
+
+        .. math::
+
+             \\xi(r) = \\frac{<D, D> - 2 f <D, R> + f^2<R, R>}{f^2<R, R>},
+
+        where the brakets represents counting pairs between two data sets
+        in a finite bin around ``r`` (distance), corresponding to setting
+        `cumulative=False`, and ``f = float(len(D)) / float(len(R))`` is the
+        ratio between number of objects from data and random.
+
+        The algorithm implemented here is loosely based on the dual-tree algorithm
+        described in [1]_. We
```

Comment:
Hmm. How do you reflow?

I have been manually breaking lines at concept groups to keep the line short. (at least attempting to)
Is the the goal to keep the lines less than 80 char wide?